node/src/actor_system_factory.rs

```diff
@@ -1251,69 +1268,26 @@ mod tests {
         assert_eq!(*add_mapping_params, vec![1234, 2345]);
     }
 
-    #[cfg(feature = "log_recipient_test")]
     #[test]
     fn prepare_initial_messages_initiates_global_log_recipient() {
-        let _guard = TEST_LOG_RECIPIENT_GUARD.lock().unwrap();
         running_test();
+        let _guard = EnvironmentGuard::new();
+        let _ = LOG_RECIPIENT_OPT.lock().unwrap().take(); // just so that it's empty
```

Comment:
This is the kind of thing which is hard to describe.

When I see this I feel a concern although I do understand your attempt to guard this test off from other tests by adding the EnvironmentalGuard. The problem is we are in the `Node/node` crate and this test could theoretically interfere with some other tests that would run the production code where the log recipient is initialized by the original design - it's code written here in this file. 

Interestingly there is another test, yes, only one, which can blow up when in collision with this one. It is `init_as_privileged_stores_dns_servers_and_passes_them_to_actor_system_factory_for_proxy_client_in_init_as_unprivileged` which is making me actually nervous because I'm seeing three various "guards" put at its beginning and it's bringing a question to me what if they could cause a deadlock. Do we need all of them? I don't know. 

Anyway. Look at this line `let _guard = TEST_LOG_RECIPIENT_GUARD.lock().unwrap();`. It's supposed to be a guard that will eliminate a possible interference with the test this comment is for. 

So, please think about the danger and you might end up with the reimplementation of `TEST_LOG_RECIPIENT_GUARD` because only so this test is going to be in harmony with the other one unless you want to do something of your own design but I think `TEST_LOG_RECIPIENT_GUARD` isn't a bad idea. 

Also notice that currently the only usage of `TEST_LOG_RECIPIENT_GUARD` in `bootstrapper.rs` is nonsense with no effect but if you pair it up with another usage here the effect will come.

Reply:
Yeah, we should probably talk more about this.

Reply:
Talked more.